pro-planner-client/src/redux/pollSlice.js

```diff
@@ -0,0 +1,49 @@
+import {createSlice} from "@reduxjs/toolkit";
+
+const pollSlice = createSlice({
+    name: 'poll',
+    initialState: {
+        polls: [],
+        votedUsers: {}
+    },
+    reducers: {
+        addPoll: (state, action) => {
+            const input = action.payload;
+            const newPoll = {
+                pollId: input.pollId,
+                question: input.question,
+                options: input.options,
+                votedUsers: input.votedUsers
+            };
+            state.polls.push(newPoll);
+        },
+        addOption: (state, action) => {
+            const input = action.payload;
+            const newOption = {
+                optionId: input.optionId,
+                option: input.option,
+                voteCount: input.voteCount
+            }
+            const poll = state.polls.find((p) => p.pollId === input.pollId);
+            poll.options.push(newOption);
+        },
+        voteOption: (state, action) => {
+            const input = action.payload;
+
+            const poll = state.polls.find((p) => p.pollId === input.pollId);
+            // const userVoted = poll.votedUsers[input.userId]; TODO... implement after userId is set
```

Comment:
Just to get something that works, you can probably use usernames as their IDs for now. Just keep in mind that it will likely get changed to using IDs, so it'd be best if you could keep any username usages to be easily refactored.